include/openthread/platform/radio.h

```diff
@@ -270,12 +270,11 @@ typedef struct otRadioFrame
          */
         struct
         {
-            const otMacKeyMaterial *mAesKey;  ///< The key material used for AES-CCM frame security.
-            otRadioIeInfo          *mIeInfo;  ///< The pointer to the Header IE(s) related information.
-            uint32_t                mTxDelay; ///< The delay time for this transmission (based on `mTxDelayBaseTime`).
-            uint32_t                mTxDelayBaseTime; ///< The base time for the transmission delay.
```

Comment:
@edmont I understand it is good optimization and from code simplification prespective, it makes perfect sense, but we generally aim to keep public and platform APIs and definitions unchanged to allow backward compatibility. 
- We can add new APIs or new fields in a `struct` but try to avoid removing or changing existing ones (unless we have a strong reason/justification).

Different radio platform implementation may be using these variables and changing this will break their code.

Reply:
Thanks @abtink, I understand. I made a quick search on platform implementations and only `ot-nrf528xx` uses both fields, the rest just add them internally.

Anyway, right now there is no strong reason for breaking backwards compatibility other than code efficiency. In any case, do you think we can leave this on hold and merge it when the time comes for introducing a non compatible change? In case we remember about it by then :)

Reply:
Thanks. Yes. Sounds good to me. If we later need to break compatibility for some good reason, make sense to use the same change to do other enhancements :)